src/com/videojs/providers/HTTPVideoProvider.as

```diff
@@ -205,16 +205,14 @@ package com.videojs.providers{
             appendBuffer(FLV_HEADER);
         }
 
-        public function get buffered():Number{
-            // _src.path == null when in data generation mode
-            if(_ns && _src.path == null)
-            {
-                return _startOffset + _ns.bufferLength + _ns.time;
-            } else if(duration > 0){
-                return (_ns.bytesLoaded / _ns.bytesTotal) * duration;
```

Comment:
So it appears bufferLength behaves completely differently in data generation mode than with PD. I restored the old calculation because (despite the fact I still think it's inaccurate) it's much better than using bufferLength.